File: Hash/camouflage.py
from collections import Counter


def solution(clothes):
    counter_each_category = Counter([cat for _, cat in clothes])
    print([cat for _, cat in clothes])  # ['headgear', 'eyewear', 'headgear']
    print(counter_each_category)  # Counter({'headgear': 2, 'eyewear': 1})

    all_possible = 1

    for key in counter_each_category:
        all_possible *= (counter_each_category[key] + 1)  # +1 : 안입는 경우를 더한 것

    return all_possible - 1  # -1 : 다 안입는 경우를 뺀 것


if __name__ == "__main__":
    print(solution([["yellow_hat", "headgear"], ["blue_sunglasses", "eyewear"], ["green_turban", "headgear"]]))

File: JS/camouflage.js
/**
 *  위장
 */

function solution(clothes) {
  let answer = 1;
  const map = new Map();

  clothes.forEach((value) => {
    if (map.has(value[1])) map.set(value[1], map.get(value[1]) + 1);
    else map.set(value[1], 1);
  });

  map.forEach((value, key) => {
    answer *= value + 1;
  });

  return answer - 1;
}
